tests/test_modeling_common.py

```diff
@@ -3167,33 +3167,85 @@ def test_mismatched_shapes_have_properly_initialized_weights(self):
         configs_no_init = _config_zero_init(config)
 
         for model_class in self.all_model_classes:
-            if model_class.__name__ not in get_values(MODEL_FOR_SEQUENCE_CLASSIFICATION_MAPPING_NAMES):
+            mappings = [
+                MODEL_FOR_SEQUENCE_CLASSIFICATION_MAPPING_NAMES,
+                MODEL_FOR_IMAGE_CLASSIFICATION_MAPPING_NAMES,
+                MODEL_FOR_AUDIO_CLASSIFICATION_MAPPING_NAMES,
+                MODEL_FOR_VIDEO_CLASSIFICATION_MAPPING_NAMES,
+            ]
+            is_classication_model = any(model_class.__name__ in get_values(mapping) for mapping in mappings)
+
+            if not is_classication_model:
                 continue
 
+            # TODO: ydshieh
+            is_special_classes = model_class.__name__ in [
+                "wav2vec2.masked_spec_embed",
+                "Wav2Vec2ForSequenceClassification",
+                "CLIPForImageClassification",
+                "RegNetForImageClassification",
+                "ResNetForImageClassification",
+            ]
+            special_param_names = [
+                r"wav2vec2\.masked_spec_embed",
+                r"wav2vec2\.feature_extractor\.conv_layers\..+\.conv\.weight",
+                r"wav2vec2\.feature_projection\.projection\.weight",
+                r"wav2vec2\.feature_projection\.projection\.bias",
+                r"wav2vec2\.encoder\.pos_conv_embed\.conv\.parametrizations\.weight\.original.",
+                r"classifier\.weight",
+                r"regnet\.embedder\.embedder\.convolution\.weight",
+                r"regnet\.encoder\.stages\..+\.layers\..+\.layer\..+\.convolution\.weight",
+                r"regnet\.encoder\.stages\..+\.layers\..+\.shortcut\.convolution\.weight",
+                r"regnet\.encoder\.stages\..+\.layers\..+\.layer\..+\.attention\..+\.weight",
+                r"regnet\.encoder\.stages\..+\.layers\..+\.layer\..+\.attention\..+\.bias",
+                r"classifier\..+\.weight",
+                r"classifier\..+\.bias",
+                r"resnet\.embedder\.embedder\.convolution\.weight",
+                r"resnet\.encoder\.stages\..+\.layers\..+\.shortcut\.convolution\.weight",
+                r"resnet\.encoder\.stages\..+\.layers\..+\.layer\..+\.convolution\.weight",
+                r"resnet\.encoder\.stages\..+\.layers\..+\.shortcut\.convolution\.weight",
+                r"resnet\.encoder\.stages\..+\.layers\..+\.layer\..+\.attention\..+\.weight",
+                r"resnet\.encoder\.stages\..+\.layers\..+\.layer\..+\.attention\..+\.bias",
+            ]
```

Comment:
So let's not skip, but allow the mean values to be in `[-1.0, 1.0]` for some parameters of certain model classes.